StripeCore/StripeCore/Source/Analytics/AnalyticLoggableError.swift

```diff
@@ -65,4 +67,79 @@ where Self: RawRepresentable, Self.RawValue == String {
         }
     }
 
+    /// Serialize an Error for logging to q.stripe.com and the `sdk.analytics_events` table
+    ///
+    /// It sends the following fields:
+    /// - error_type: For Stripe API errors, the error’s [type](https://docs.stripe.com/api/errors#errors-type) e.g. “invalid_request_error”.
+    ///           For Swift errors, the fully qualified type name e.g. “StripePaymentSheet.LinkURLGeneratorError”.
+    ///           For NSErrors, the error domain e.g. “NSURLErrorDomain”.
+    /// - error_code: For Stripe API errors, the error's code e.g. "invalid_number".
+    ///            For NSErrors, the error code e.g. “-1009”.
+    ///            For Swift errors, the enum case name as a string for Swift errors e.g. “noPublishableKey”.
+    public func serializeForV1Analytics() -> [String: Any] {
+        let errorType = Self.extractErrorType(from: self)
+        let errorCode = Self.extractErrorCode(from: self)
+        
+        return [
+            "error_type": errorType,
+            "error_code": errorCode
+        ]
+    }
+    
+    /// Extracts a value suitable for the `"error_type"` analytic parameter
+    /// - For Stripe API errors, the error’s [type](https://docs.stripe.com/api/errors#errors-type) e.g. “invalid_request_error”.
+    /// - For Swift errors, the fully qualified type name e.g. “StripePaymentSheet.LinkURLGeneratorError”.
+    /// - For NSErrors, the error domain e.g. “NSURLErrorDomain”.
+    static func extractErrorType(from error: Error) -> String {
+        if type(of: error) is NSError.Type {
+            let error = error as NSError
+            if error.domain == STPError.stripeDomain, let stripeAPIErrorType = error.userInfo[STPError.stripeErrorTypeKey] as? String {
+                // For Stripe API Error, use the error type key's value
+                return stripeAPIErrorType
+            } else {
+                // Default behavior for other errors.
+                // Note: For Swift Error enums, `domain` is the type name
+                // e.g. `LinkURLGeneratorError.noPublishableKey` -> "StripePaymentSheet.LinkURLGeneratorError"
+                return "\(error.domain)"
+            }
+        } else {
+            return String(reflecting: type(of: error))
+        }
+    }
+    
+    /// Extracts a value suitable for the `"error_code"` analytic parameter
+    /// - For Stripe API errors, the error's code e.g. "invalid_number".
+    /// - For NSErrors, the error code e.g. “-1009”.
+    /// - For Swift errors, the enum case name as a string for Swift errors e.g. “noPublishableKey”.
+    static func extractErrorCode(from error: Error) -> String {
```

Comment:
The code is pretty complicated :(. Turns out it's really hard to get details about Errors if you have to avoid invoking "description" or "debugDescription"!  And relying on `(error as NSError).code` leads to confusing values if the enum has associated values, which is a common thing - the cases with associated values seem to get ordered first. For example:
```
Enum MyError: Error {
 Case zero
 Case one(foo: String)
}
print((MyError.zero as NSError).code) // 1
print((MyError.one(foo: “”) as NSError).code) // 0
```

I'm really open to ideas to improve it. If it's too gnarly we can just use `(error as NSError).code` and deal with the confusing values.